### ISSUE TITLE
Issues with using/re-training saved models

### ISSUE THREAD

**Issue by Felicity** [MicrosoftResearch/Azimuth#7]
Hi,

I'm trying to compute Rule Set 2 scores using model_comparison.predict on the nopos model. 

If I download the latest version from github, the pickle files don't seem to load:

```
ValueError: Buffer dtype mismatch, expected 'SIZE_t' but got 'long long'
```

So I run model_comparison.py as instructed on the main github page to reproduce the two models.
It prints a LOT of warnings:

```
WARNING: trimming max_index_to use down to length of string=30
```

but does eventually produce two model files.

However if I use those files, I get a different answer to the value you report in the README that comes with the Rule Set 2 Calculator 0.5909 vs 0.5656.

Looking at that latest test_saved_models.py, it looks like you expect the new value. So I guess I'm just wondering what the reason is for the change? Which one corresponds to the method used in your paper? I'm guessing the older one? In which case, how do I reproduce that? 

(sorry I didn't mean to submit this issue when I submitted it, and have since resolved some issues with versioning, so now it's a question instead!)

Thanks,
Felicity

**Comment by Felicity** [comment]
A further observation is that the web server appears to produce the 0.5656 answer.

**Comment by nfusi** [comment]
Hi Felicity,

> If I download the latest version from github, the pickle files don't seem to load

you could be using the wrong version of scikit-learn. If you install Azimuth now, it should force you to use the correct version. You can also retrain the models to match your version of scikit-learn (see instructions in the readme).

> So I guess I'm just wondering what the reason is for the change? 

We fixed a small indexing bug in #1, so perhaps that's the reason for the discrepancy. You can find more details in release v1.0 [here](https://github.com/MicrosoftResearch/Azimuth/releases).

Let us know if any of this helps, I will close the bug for now.

Nicolo
